tmpfiles.sh

```diff
@@ -368,11 +352,7 @@ _r() {
 			dryrun_or_real rm -f "$path"
 		elif [ -d "$path" ]; then
 			dryrun_or_real rmdir "$path"
-		fi
-		x=$?
-		if [ $x -ne 0 ]; then
-			status=$x
-		fi
+		fi || status="$?"
```

Comment:
please merge this into the individual dryrun_or_real calls above.  i know this will duplicate, but the code will be easier to understand for people.  this style of code has also been unreliable in some shells (handling of $? across `if` boundaries), although it prob doesn't apply to dash/bash.